File: core/src/test/java/de/qabel/core/accounting/TestAccountingServerBuilder.java
package de.qabel.core.accounting;

import de.qabel.core.TestServer;
import de.qabel.core.config.AccountingServer;

import java.net.URI;
import java.net.URISyntaxException;

public class TestAccountingServerBuilder {
    public static final String DEFAULT_PASS = "testuser";
    private URI accountingUri;
    private URI blockUri;
    private String user = "testuser";
    private String pass = DEFAULT_PASS;

    public TestAccountingServerBuilder() throws URISyntaxException {
        accountingUri = new URI(TestServer.ACCOUNTING);
        blockUri = new URI(TestServer.BLOCK);
    }

    public TestAccountingServerBuilder user(String user) {
        this.user = user;
        if (pass.equals(DEFAULT_PASS)) {
            pass = "randomPass";
        }
        return this;
    }

    public TestAccountingServerBuilder pass(String pass) {
        this.pass = pass;
        return this;
    }

    public TestAccountingServerBuilder block(URI blockUri) {
        this.blockUri = blockUri;
        return this;
    }

    public TestAccountingServerBuilder accounting(URI accountingUri) {
        this.accountingUri = accountingUri;
        return this;
    }

    public AccountingServer build() {
        return new AccountingServer(
            accountingUri,
            blockUri,
            user,
            pass
        );
    }
}

File: core/src/test/java/de/qabel/core/accounting/TestAccountingServerBuilder.kt
package de.qabel.core.accounting

import de.qabel.core.config.AccountingServer

import java.net.URI
import java.net.URISyntaxException

class TestAccountingServerBuilder @Throws(URISyntaxException::class)
constructor() {
    private var accountingUri: URI? = null
    private var blockUri: URI? = null
    private var user = "testuser"
    private var pass = DEFAULT_PASS

    init {
        accountingUri = URI("http://localhost:9696")
        blockUri = URI("http://localhost:9697")
    }

    fun user(user: String): TestAccountingServerBuilder {
        this.user = user
        if (pass == DEFAULT_PASS) {
            pass = "randomPass"
        }
        return this
    }

    fun pass(pass: String): TestAccountingServerBuilder {
        this.pass = pass
        return this
    }

    fun block(blockUri: URI): TestAccountingServerBuilder {
        this.blockUri = blockUri
        return this
    }

    fun accounting(accountingUri: URI): TestAccountingServerBuilder {
        this.accountingUri = accountingUri
        return this
    }

    fun build(): AccountingServer {
        return AccountingServer(
                accountingUri,
                blockUri!!,
                user,
                pass)
    }

    companion object {
        val DEFAULT_PASS = "testuser"
    }
}
